futures-util/src/async_await/join.rs

```diff
@@ -47,3 +47,77 @@ macro_rules! join {
         )*)
     } }
 }
+
+/// Polls multiple futures simultaneously, returning a [`Result`] containing
+/// either a tuple of the successful outputs or an error.
```

Comment:
"return" is IMO the wrong word because the result is inside a future

Reply:
Done.